File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i=0
        counter=0
        for j in range(len(nums)):
            if nums[j]!=0:
                nums[i]=nums[j]
                i+=1
            else:
                counter+=1
            # print(nums)
        
        for j in range(counter):
            nums[len(nums)-1-j]=0
        # print(nums)
       

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l=0;
        int counter=0;
        for(int r=0;r<nums.length;r++){
            if(nums[r]!=0){
                nums[l]=nums[r];
                l++;
            }
            else{
                counter++;
            }
        }
        for(int i=0;i<counter;i++){
            nums[nums.length-1-i]=0;
        }
    }
}